src/OpenTelemetry/Trace/Builder/TracerProviderBuilderExtensions.cs

```diff
@@ -24,6 +24,9 @@ public static class TracerProviderBuilderExtensions
     /// <param name="tracerProviderBuilder"><see cref="TracerProviderBuilder"/>.</param>
     /// <param name="enabled">Enabled or not. Default value is <c>true</c>.</param>
     /// <returns>Returns <see cref="TracerProviderBuilder"/> for chaining.</returns>
+#if NET7_0_OR_GREATER
+    [RequiresDynamicCode("Calling this method will invoke Marshal.GetExceptionPointers(), which is neither supported in MonoRuntime nor in nativeAOT environment.")]
```

Comment:
I feel "will invoke Marshal.GetExceptionPointers()" is the internal detail that we don't want to include. Consider making it simple, e.g. "SetErrorStatusOnException is not supported in MonoRuntime nor in nativeAOT environment." (take a look at the .NET runtime examples on how to phrase/communicate it properly)

Reply:
The custom message from the attribute is going to be prefixed by generic error message `Using member '...' which has 'RequiresDynamicCodeAttribute' can break functionality when AOT compiling.`. We do not mention native AOT nor Mono in these custom messages in .NET runtime libraries.

Reply:
Please see this discussion: https://github.com/open-telemetry/opentelemetry-dotnet/pull/5374#discussion_r1506907797. Thanks!

Reply:
@eerhardt Do you know if `Marshal.GetExceptionPointers()` is trim-safe?

Reply:
It's not marked as "RequiresUnreferencedCode", so my assumption would be it works in a "normal" trimmed app just fine. (It is just native code that is part of the CoreCLR runtime.) The issue with native AOT is that it is a different runtime and that runtime doesn't have the necessary native code.